### PR TITLE
Add camel-olingo4 wrapper

### PR DESCRIPTION
**Pull request by lopushen**
Left the TODO as the issue in the TODO is not resolved yet
Updated Aalto-xml to [1.3.2](https://mvnrepository.com/artifact/com.fasterxml/aalto-xml/1.3.2) as no version requirement was present and the 0.9.10 is 9 years old

### CODE REVIEW

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -1617,42 +1617,38 @@
...
+        <bundle dependency='true'>mvn:org.codehaus.woodstox/stax2-api/3.1.4</bundle>
+        <bundle dependency='true'>mvn:commons-io/commons-io/2.15.1</bundle>
+        <bundle dependency='true'>mvn:org.apache.commons/commons-lang3/${commons-lang3-version}</bundle>
+        <bundle dependency='true'>mvn:org.apache.httpcomponents/httpasyncclient-osgi/4.1.5</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-core/2.16.2</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-databind/2.16.2</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-annotations/2.16.2</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-xml/2.16.2</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.module/jackson-module-jaxb-annotations/2.16.2</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: Please use the Jackson feature

Comment (lopushen) [reply features/src/main/feature/camel-features.xml]: done, replaced the 
```
 <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-core/2.16.2</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-databind/2.16.2</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-annotations/2.16.2</bundle>
```
with the jackson feature, set the 2 latter dependencies to 2.16.1 to comply with the feature's bundles

File: features/src/main/feature/camel-features.xml [hunk 2]
```diff
@@ -1617,42 +1617,36 @@
...
-<!--    </feature>-->
+    <feature name='camel-olingo4' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
+        <feature version="[4,5)">http-client</feature>
+        <feature>jakarta-servlet</feature>
+        <feature>jackson</feature>
+        <bundle dependency='true'>mvn:org.apache.olingo/odata-commons-api/5.0.0</bundle>
+        <bundle dependency='true'>mvn:org.apache.olingo/odata-commons-core/5.0.0</bundle>
+        <bundle dependency='true'>mvn:org.apache.olingo/odata-client-api/5.0.0</bundle>
+<!--&lt;!&ndash;        &lt;!&ndash;TODO remove wrap protocol when https://issues.apache.org/jira/browse/OLINGO-1206 is resolved&ndash;&gt;&ndash;&gt;-->
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: Let's fix this comment or remove it.
It should be `<!-- TODO remove wrap protocol when https://issues.apache.org/jira/browse/OLINGO-1206 is resolved -->`

Comment (lopushen) [reply features/src/main/feature/camel-features.xml]: fixed, thanks

File: features/src/main/feature/camel-features.xml [hunk 3]
```diff
@@ -1617,42 +1617,36 @@
...
+        <bundle dependency='true'>mvn:org.apache.olingo/odata-client-api/5.0.0</bundle>
+        <!-- TODO remove wrap protocol when https://issues.apache.org/jira/browse/OLINGO-1206 is resolved -->
+        <bundle dependency='true'>wrap:mvn:org.apache.olingo/odata-server-api/5.0.0$overwrite=merge&amp;Export-Package=org.apache.olingo.server.api;version=5.0.0,org.apache.olingo.server.api.uri;version=5.0.0,org.apache.olingo.server.api.processor;version=5.0.0,org.apache.olingo.server.api.batch;version=5.0.0,org.apache.olingo.server.api.debug;version=5.0.0,org.apache.olingo.server.api.deserializer;version=5.0.0,org.apache.olingo.server.api.deserializer.batch;version=5.0.0,org.apache.olingo.server.api.etag;version=5.0.0,org.apache.olingo.server.api.prefer;version=5.0.0,org.apache.olingo.server.api.serializer;version=5.0.0,org.apache.olingo.server.api.uri.queryoption;version=5.0.0,org.apache.olingo.server.api.uri.queryoption.apply;version=5.0.0,org.apache.olingo.server.api.uri.queryoption.expression;version=5.0.0,org.apache.olingo.server.api.uri.queryoption.search;version=5.0.0,org.apache.olingo.netty.server.api;version=5.0.0,org.apache.olingo.netty.server.core;version=5.0.0</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: Could be simplified using the character *, I mean something like `Export-Package=org.apache.olingo.server.api.*;version=5.0.0`

Comment (lopushen) [reply features/src/main/feature/camel-features.xml]: done, 14 packages to one, thanks!

**Review by essobedo — APPROVED** [review]

**Pull request merged**